tests/rbd_mirror/test_rbd_group_consistency.py

```diff
@@ -0,0 +1,407 @@
+                "md5sums of images on site B after force promote on site-B: %s",
+                md5sum_after_force_promote_site_b,
+            )
+
+            # Due to existing BZ Bug 2363632 on 8.1, the force promote as of now
+            # rollsback the images to the n-1 group snapshot. Hence,
+            # md5sum_after_force_promote_site_b is compared with md5sum_first_sync_site_b
+            # as of now. This BZ is planned to be fixed in 8.1z releases. So, in 8.1z
+            # md5sum_after_force_promote_site_b should be compared with md5sum_second_sync_site_b.
+            # the below line should look something like :
+            # if md5sum_after_force_promote_site_b != md5sum_second_sync_site_b:
+            if md5sum_after_force_promote_site_b != md5sum_first_sync_site_b:
+                raise Exception(
+                    "md5sums after force promote are not consistent with previous snapshot \n"
+                    f"site-B: {md5sum_after_force_promote_site_b} \n"
+                    f"site-B: {md5sum_first_sync_site_b}"
+                )
+            exec_cmd(
+                node=client_secondary,
+                cmd="ceph orch start rbd-mirror",
+            )
+            log.info("Restored rbd-mirror daemon on site-B")
+            time.sleep(5)
+
+
+def run(**kw):
+    """
+    This test verifies group level data consistency on secondary after primary site disaster
+    Args:
+        kw: test data
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    """
+    try:
+        pool_types = ["rep_pool_config", "ec_pool_config"]
+        grouptypes = ["single_pool_without_namespace", "single_pool_with_namespace"]
+
+        if not kw.get("config").get("grouptype"):
+            for pooltype in pool_types:
+                group_type = grouptypes.pop(random.randrange(len(grouptypes)))
+                kw.get("config").get(pooltype).update({"grouptype": group_type})
+                log.info("Choosing Group type on %s - %s", pooltype, group_type)
+        mirror_obj = initial_mirror_config(**kw)
+        mirror_obj.pop("output", [])
+        for val in mirror_obj.values():
+            if not val.get("is_secondary", False):
+                rbd_primary = val.get("rbd")
+                client_primary = val.get("client")
+                primary_cluster = val.get("cluster")
+            else:
+                rbd_secondary = val.get("rbd")
+                client_secondary = val.get("client")
+                secondary_cluster = val.get("cluster")
+
+        pool_types = list(mirror_obj.values())[0].get("pool_types")
+        test_group_consistency(
+            rbd_primary,
+            rbd_secondary,
+            client_primary,
+            client_secondary,
+            primary_cluster,
+            secondary_cluster,
+            pool_types,
+            **kw,
+        )
+        log.info(
+            "Test Verify group level data consistency on secondary after primary site disaster passed"
```

Comment:
It should be
`The test verifying group-level data consistency on the secondary site after a primary site disaster has passed.`